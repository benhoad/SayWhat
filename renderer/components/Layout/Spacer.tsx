import cn from "classnames";

interface ISpacerProps {
  expand?: boolean;
  children?: any;
}

export default function Spacer({ children, expand = false }: ISpacerProps) {
  return (
    <>
      <div className={cn("Spacer", { Expand: expand })}>{children}</div>

      <style jsx>{`
        .Spacer {
          min-width: 2rem;
        }
        .Spacer.Expand {
          flex-grow: 1;
          width: 100%;
        }
      `}</style>
    </>
  );
}
