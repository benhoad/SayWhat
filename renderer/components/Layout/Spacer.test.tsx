import { render } from "@testing-library/react";

import Spacer from "./Spacer";

describe("Spacer", () => {
  it("renders", () => {
    expect.hasAssertions();

    const { container } = render(<Spacer />);
    expect(container.firstElementChild).toMatchSnapshot();
  });
});
