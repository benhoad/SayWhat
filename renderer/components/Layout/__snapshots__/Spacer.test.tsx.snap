// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer renders 1`] = `
<div
  class="jsx-1127755532 Spacer"
/>
`;
